lib/views/home_content/map_feed/map_feed.dart

```diff
@@ -0,0 +1,71 @@
+import "package:flutter/material.dart";
+import "package:google_maps_flutter/google_maps_flutter.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/services/geolocation_service.dart";
+import "package:proxima/views/home_content/map_feed/maps/nearby_posts_map.dart";
+import "package:proxima/views/sort_option_widgets/feed_sort_option/map_sort_option_chips.dart";
+
+class MapFeed extends ConsumerStatefulWidget {
+  const MapFeed({super.key});
+
+  @override
+  MapFeedState createState() => MapFeedState();
+}
```

Comment:
I think that this is a viewmodel. Have you tried to implement it using Riverpod ? (Maybe Google map require this type but I think it is possible https://dev.to/xxxuutaxxx/a-quick-guide-googlemapsflutter-with-current-location-4l4k) . 

Moreover we should move the viewmodel into the viewmodel folder.